frontend/src/app/modules/social-page/social-page/social-page.component.ts

```diff
@@ -13,16 +14,61 @@ import { UserFilter } from '../../../models/filters/userFilter';
 export class SocialPageComponent implements OnInit {
     public users$: Observable<UserCard[]>;
 
+    public slicedUsers$: Observable<UserCard[][]>;
+
+    currentPage: number = 0;
+
+    paginationPages: number = 10;
+
+    public userCount$: Observable<number>;
+
     public constructor(
         private userService: UserService,
     ) {
     }
 
     ngOnInit(): void {
         this.users$ = this.userService.getUsers();
+
+        this.userCount$ = this.users$.pipe(
+            map(users => users.length),
+        );
+        this.slicedUsers$ = this.users$.pipe(
+            map((users) => this.chunkArray(users, this.paginationPages)),
+        );
     }
 
     onFilterChanges(filters: UserFilter) {
         this.users$ = this.userService.getUsers(filters);
+
+        this.userCount$ = this.users$.pipe(
+            map(users => users.length),
+        );
+
+        this.slicedUsers$ = this.users$.pipe(
+            map((users) => this.chunkArray(users, this.paginationPages)),
+        );
     }
+
+    onChanged(numberVar: number) {
+        this.currentPage = numberVar - 1;
```

Comment:
Can't you make it like in one format, so it will be everywhere 1,2,3,4... or 0,1,2,3...